### DISCUSSION TITLE
Eulerian-Eulerian (2-Phase) kernals

### DISCUSSION THREAD

**Discussion by AnthonyB08** [idaholab/moose#23530] [Q&A Modules: Navier-Stokes]
Hi all,

Is it possible to model an Eulerian-Eulerian (gas-liquid) fluid flow in MOOSE? What kind of kernels would I look towards?

**Comment by GiudGiud** [comment]
Hello

No we do not have two phase (gas-liquid) fluid flow in MOOSE. 
Are you doing CFD or 1D thermal hydraulics? For the latter you can request access to RELAP-7. 
For CFD you could develop your own set of kernels and we would be interested in this contribution

Guillaume

**Reply by AnthonyB08** [reply]
I am doing CFD with OPENMC and MOOSE wrapped by Cardinal. I am interested in modeling the production and removal of xenon-135 by sparging helium bubbles. This was easily accomplished in ANSYS Fluent, but lacked the coupling of neutronics and thermals. 

Now, I want to take my work a step further by incorporating neutronics, thermal, and fluid flow into this topic. My draft "idea" is a simple 2-D closed loop with the core on one side and components on the opposing side. A momentum source term that acts as a pump and a sink term for the heat exchanger. Xenon-135 will be a source term in the continuity equation, which is coupled with neutronics from OPENMC.

I am leaning towards the ladder since it may also fulfill my dissertation topic. However, I have a few gaps in my understanding of MOOSE and Bubble-bubble CFD modeling that I need to fulfill prior to implementation. 

-Thanks

**Reply by GiudGiud** [reply]
Sounds good. Please let us know if you run into any issue implementing or have any questions. 

You may consider a simple drift flux model as a first step.
having the Xenon concentration as a scalar quantity in solution should be fairly easy to do from existing kernels. Maybe looking at this MSFR model where this is done for precursors will help
https://github.com/idaholab/virtual_test_bed/blob/devel/msr/msfr/steady/run_ns.i
